mvcc/key_index.go

```diff
@@ -210,18 +223,21 @@ func (ki *keyIndex) compact(atRev int64, available map[revision]struct{}) {
 
 	if !g.isEmpty() {
 		n := g.walk(f)
-		// remove the previous contents.
-		if n != -1 {
-			g.revs = g.revs[n:]
-		}
 		// remove any tombstone
-		if len(g.revs) == 1 && i != len(ki.generations)-1 {
+		if len(g.revs)-1 == n && i != len(ki.generations)-1 {
```

Comment:
swapping the order of the if's doesn't look like it preserves the meaning; if `n > 0` then `delete(available, g.revs[0])` is not the same as the current code